# 1. Inspección Inicial del Dataset
En esta sección realizamos la carga del dataset original en su formato crudo y una primera exploración de su estructura, dimensiones y tipos de datos para reunir evidencia inicial.

In [1]:
import pandas as pd

df = pd.read_json("../data/raw/streaming_users_dirty.json")

df.head()

,user_id,age,subscription_plan,monthly_watch_time_mins,country,favorite_genre,last_login_date,customer_support_tickets
0,10000,39,Estándar,805.8,Brasil,Crime,2025-03-04,99
1,10001,37,Estándar,1173.4,Colombia,Crime,2019-04-02,2
2,10002,28,Básico,401.0,Colombia,Crime,2018-04-13,0
3,10003,43,Básico,62.4,Uruguay,Thriller,2021-01-31,0
4,10004,51,Básico,477.8,Perú,Thriller,2020-09-30,1



Verificamos la cantidad total de registros (filas) y variables (columnas) que componen el conjunto de datos.

In [2]:
df.shape

(8160, 8)


Analizamos los tipos de datos asignados a cada columna y verificamos la consistencia general de la estructura.

In [3]:
df.info()

<class 'pandas.DataFrame'>
RangeIndex: 8160 entries, 0 to 8159
Data columns (total 8 columns):
 #   Column                    Non-Null Count  Dtype  
---  ------                    --------------  -----  
 0   user_id                   8160 non-null   int64  
 1   age                       8160 non-null   int64  
 2   subscription_plan         8160 non-null   str    
 3   monthly_watch_time_mins   7967 non-null   float64
 4   country                   8160 non-null   str    
 5   favorite_genre            7920 non-null   str    
 6   last_login_date           7840 non-null   str    
 7   customer_support_tickets  8160 non-null   int64  
dtypes: float64(1), int64(3), str(4)
memory usage: 756.7 KB



Contabilizamos la cantidad de valores nulos presentes en cada una de las variables para orientar la posterior etapa de calidad y limpieza.

In [4]:
df.isnull().sum()

user_id                       0
age                           0
subscription_plan             0
monthly_watch_time_mins     193
country                       0
favorite_genre              240
last_login_date             320
customer_support_tickets      0
dtype: int64


Verificamos si existen filas idénticas repetidas dentro del dataset.

In [5]:
df.duplicated().sum()

np.int64(126)

 Conclusiones de la Inspección Inicial
A partir de la evidencia recolectada, se detectaron las siguientes particularidades en el dataset crudo:
* **Registros Duplicados:** Se identificaron exactamente 126 filas completamente duplicadas que deberán ser eliminadas en la siguiente etapa para evitar sesgos en el análisis.
* **Valores Faltantes:** Se observaron valores nulos en columnas clave (ver output de la celda 4), lo que requerirá definir una estrategia de imputación o eliminación justificada en el próximo notebook.

Con este diagnóstico, el dataset está listo para el proceso de calidad y limpieza en `02_calidad_y_limpieza.ipynb`.